spec/requests/maintenance_controller_spec.rb

```diff
@@ -0,0 +1,54 @@
+require "rails_helper"
+
+RSpec.describe MaintenanceController, type: :request do
+  let(:page) { Capybara::Node::Simple.new(response.body) }
+  let(:user) { FactoryBot.create(:user) }
+
+  before do
+    sign_in user
+  end
+
+  describe "when maintenance mode is enabled" do
+    before do
+      allow(FeatureToggle).to receive(:maintenance_mode_enabled?).and_return(true)
+    end
+
+    context "when a user visits a page other than the maintenance page" do
+      it "redirects the user to the maintenance page" do
+        get "/lettings-logs"
+        expect(response).to redirect_to(service_unavailable_path)
+        follow_redirect!
+        expect(page).to have_content("Sorry, the service is unavailable")
```

Comment:
I haven't tested that the user is signed out here, because I tested that in `spec/controllers/maintenance_controller_spec.rb`.